api/base/serializers.py

```diff
@@ -177,8 +234,27 @@ def to_representation(self, obj, envelope='data'):
         meta = getattr(self, 'Meta', None)
         type_ = getattr(meta, 'type_', None)
         assert type_ is not None, 'Must define Meta.type_'
-        data = super(JSONAPISerializer, self).to_representation(obj)
-        data['type'] = type_
+
+        data = collections.OrderedDict([('id', ''), ('type', type_), ('attributes', collections.OrderedDict()),
+                                        ('relationships', collections.OrderedDict()), ('links', {})])
+
+        fields = [field for field in self.fields.values() if not field.write_only]
+
+        for field in fields:
+            try:
+                attribute = field.get_attribute(obj)
+            except SkipField:
+                continue
+
+            if isinstance(field, HyperlinkedIdentityFieldWithMeta):
```

Comment:
I've also been working on DraftRegistrations on the API side...HyperlinkedRelatedField works well there because we wanted to return links to branched_from and initiator, both of which are detail views.

![screen shot 2015-09-01 at 4 57 50 pm](https://cloud.githubusercontent.com/assets/9755598/9623029/57a23990-5109-11e5-9374-36e871ba1260.png)